symengine/logic.h

```diff
@@ -96,6 +96,71 @@ inline RCP<const Basic> piecewise(PiecewiseVec &&vec)
     return make_rcp<Piecewise>(std::move(vec));
 }
 
+class And : public Boolean
+{
+public:
+    set_boolean args;
+
+public:
+    IMPLEMENT_TYPEID(AND)
+    And(const set_boolean &s);
+    bool is_canonical(const set_boolean &container);
+    //! \return the hash
+    std::size_t __hash__() const;
+    virtual vec_basic get_args() const;
+    virtual bool __eq__(const Basic &o) const;
+    //! Structural equality comparator
+    virtual int compare(const Basic &o) const;
+};
+
+class Or : public Boolean
+{
+public:
+    set_boolean args;
+
+public:
+    IMPLEMENT_TYPEID(OR)
+    Or(const set_boolean &s);
+    bool is_canonical(const set_boolean &container);
+    //! \return the hash
+    std::size_t __hash__() const;
+    virtual vec_basic get_args() const;
+    virtual bool __eq__(const Basic &o) const;
+    //! Structural equality comparator
+    virtual int compare(const Basic &o) const;
+};
+
+class Not : public Boolean
+{
+public:
+    RCP<const Boolean> arg;
+
+public:
+    IMPLEMENT_TYPEID(NOT)
+    Not(const RCP<const Boolean> &s);
+    bool is_canonical(const RCP<const Boolean> &s);
+    //! \return the hash
+    std::size_t __hash__() const;
+    virtual vec_basic get_args() const;
+    virtual bool __eq__(const Basic &o) const;
+    //! Structural equality comparator
+    virtual int compare(const Basic &o) const;
+};
+
+template <typename caller>
+RCP<const Boolean> and_or(const set_boolean &s, const bool &op_x_notx);
```

Comment:
You have to move this to the implementation file as it is not a good idea to keep only the definition of a templated functions in a header file.